sdk/core/azure-core/inc/azure/core/internal/tracing/service_tracing.hpp

```diff
@@ -215,6 +215,22 @@ namespace Azure { namespace Core { namespace Tracing { namespace _internal {
                   ->CreateTracer(packageName, packageVersion);
       }
     }
+    /**
+     * @brief Construct a new Tracing Context Factory object
+     *
+     * @param options Client Options for tracing.
+     * @param serviceName Name of the resource provider for the service [See
+     * also](https://docs.microsoft.com/azure/azure-resource-manager/management/azure-services-resource-providers).
+     * @param packageVersion Optional package version number for the package containing this
+     * service. (https://opentelemetry.io/docs/reference/specification/trace/api/#get-a-tracer).
+     */
+    [[deprecated]] TracingContextFactory(
+        Azure::Core::_internal::ClientOptions const& options,
+        std::string const& serviceName,
+        std::string packageVersion)
+        : TracingContextFactory(options, serviceName, serviceName, packageVersion)
```

Comment:
nit: keep the same parameter names as before (and update the doc comment)

`packageVersion` -> `serviceVersion`

https://github.com/Azure/azure-sdk-for-cpp/blob/0e9fba2623a7c7d4625573e46809e7e358323be9/sdk/core/azure-core/inc/azure/core/internal/tracing/service_tracing.hpp#L191-L194

Reply:
The new names correctly express the roles of the parameters, the old names did *not* reflect the roles of the parameters. Yes, it's deprecated, but the parameters IMHO should still be accurate.

Reply:
Do we characterize an argument rename to be breaking? I know in Embedded C, it isn't, but we do have named parameters in C++.

Given we are deprecating an existing API (to avoid breaks), I would imagine we leave the "poor" name, to avoid any breaks.

Tha said, since it is internal, and we control the callsites (and we can audit whether we use a named parameter or not), it seems OK to change the parameter name here. It just seems unnecessary work to do so :)